src/lazy/text/buffer.rs

```diff
@@ -1428,6 +1645,50 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_match_timestamp() {
+        fn match_timestamp(input: &str) {
+            MatchTest::new(input).expect_match(match_length(TextBufferView::match_timestamp));
+        }
+        fn mismatch_timestamp(input: &str) {
+            MatchTest::new(input).expect_mismatch(match_length(TextBufferView::match_timestamp));
+        }
+
+        let good_inputs = &[
+            "2023T",
+            "2023-08T",
+            "2023-08-13", // T is optional for ymd
+            "2023-08-13T",
+            "2023-08-13T14:18Z",
+            "2023-08-13T14:18+05:00",
+            "2023-08-13T14:18-05:00",
+            "2023-08-13T14:18:35-05:00",
+            "2023-08-13T14:18:35.994-05:00",
+        ];
+        for input in good_inputs {
+            match_timestamp(input);
+        }
+
+        let bad_inputs = &[
+            "2023",                       // No 'T'
+            "2023-08",                    // No 'T'
+            "20233T",                     // 5-digit year
+            "2023-13T",                   // Out of bounds month
+            "2023-08-41T",                // Out of bounds day
+            "2023-08+18T",                // Wrong delimiter
+            "2023-08-18T25:00Z",          // Out of bounds hour
+            "2023-08-18T14:00",           // No offset
+            "2023-08-18T14:62",           // Out of bounds minute
+            "2023-08-18T14:35:61",        // Out of bounds second
+            "2023-08-18T14:35:52.Z",      // Dot but no fractional
+            "2023-08-18T14:35:52.+24:30", // Out of bounds offset hour
+            "2023-08-18T14:35:52.+00:60", // Out of bounds offset minute
```

Comment:
These both have a dot but no fractional digits, so the test case can pass even if the out of bounds offset is not correctly rejected.